instrumentation/jetty-httpclient/jetty-httpclient-12.0/library/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v12_0/TracingHttpRequest.java

```diff
@@ -37,9 +41,85 @@ public void send(Response.CompleteListener listener) {
     // start span and attach listeners.
     JettyClientTracingListener.handleRequest(parentContext, this, instrumenter);
     super.send(
-        result -> {
-          try (Scope scope = openScope()) {
-            listener.onComplete(result);
+        new Response.Listener() {
+          @Override
+          public void onBegin(Response response) {
+            if (listener instanceof Response.BeginListener l) {
+              try (Scope scope = openScope()) {
+                l.onBegin(response);
+              }
+            }
+          }
+
+          @Override
+          public void onComplete(Result result) {
+            try (Scope scope = openScope()) {
+              listener.onComplete(result);
+            }
+          }
+
+          @Override
+          public void onContent(Response response, ByteBuffer content) {
+            if (listener instanceof Response.ContentListener l) {
+              try (Scope scope = openScope()) {
+                l.onContent(response, content);
+              }
+            }
+          }
+
+          @Override
+          public void onContent(Response response, Chunk chunk, Runnable demander) {
+            if (listener instanceof Response.AsyncContentListener l) {
+              try (Scope scope = openScope()) {
+                l.onContent(response, chunk, demander);
+              }
+            }
+          }
+
+          @Override
+          public void onContentSource(Response response, Source contentSource) {
+            if (listener instanceof Response.ContentSourceListener l) {
+              try (Scope scope = openScope()) {
+                l.onContentSource(response, contentSource);
+              }
+            }
+          }
```

Comment:
The failing tests https://scans.gradle.com/s/gak4jmkou2xve/tests/task/:instrumentation:jetty-httpclient:jetty-httpclient-12.0:library:test/overview start to pass after removing these 3 listeners. Perhaps it would be better to use something similar to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/internal/JettyClientWrapUtil.java to make the wrapped listener only implement the interfaces that the original listener has.
Also you should add a comment explaining why this is needed.
Consider adding a test.

Reply:
Thanks @laurit for the feedback. I updated the approach, re-using the wrapper with some minor simplifications. What do you think of the new version? Thanks!